packages/server/lib/StudioLifecycleManager.ts

```diff
@@ -0,0 +1,151 @@
+import { EventEmitter } from 'stream'
+import type { StudioManager } from './cloud/studio'
+import { ProtocolManager } from './cloud/protocol'
+import { getAndInitializeStudioManager } from './cloud/api/get_and_initialize_studio_manager'
+import Debug from 'debug'
+import type { CloudDataSource } from '@packages/data-context/src/sources'
+import type { Cfg } from './project-base'
+import _ from 'lodash'
+import type { DataContext } from '@packages/data-context'
+import api from './cloud/api'
+
+const debug = Debug('cypress:server:studio-lifecycle-manager')
+const routes = require('./cloud/routes')
+
+export class StudioLifecycleManager extends EventEmitter {
+  studioManagerPromise: Promise<StudioManager | null> | null = null
+  private static readonly STUDIO_READY_EVENT = 'studio:ready'
+  private ctx: DataContext | null = null
+
+  /**
+   * Set the studio manager promise and register listeners
+   * @param studioManagerPromise Promise that resolves to a StudioManager
+   */
+  private setStudioManagerPromise (studioManagerPromise: Promise<StudioManager | null>): void {
+    this.studioManagerPromise = studioManagerPromise
+
+    // When the promise resolves, emit the studio:ready event with the studio manager
+    void studioManagerPromise.then((studioManager) => {
+      this.emit(StudioLifecycleManager.STUDIO_READY_EVENT, studioManager)
+    })
+  }
+
+  /**
+   * Initialize the studio manager and possibly set up protocol.
+   * Also registers this instance in the data context.
+   * @param projectId The project ID
+   * @param cloudDataSource The cloud data source
+   * @param cfg The project configuration
+   * @param debugData Debug data for the configuration
+   * @param ctx Data context to register this instance with
+   */
+  async initializeStudioManager ({
+    projectId,
+    cloudDataSource,
+    cfg,
+    debugData,
+    ctx,
+  }: {
+    projectId?: string
+    cloudDataSource: CloudDataSource
+    cfg: Cfg
+    debugData: any
+    ctx: DataContext
+  }): Promise<void> {
```

Comment:
Does this need to be `Promise<void>` and an async function? We're not actually awaiting anything. If we do that, I think we can drop the `void` in `project-base`